test/backend_mlir.in.cpp

```diff
@@ -0,0 +1,250 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+// End to end test for MLIR. Add tests here that are specific to test MLIR functionality
+// MLIR is implicitly tested during other unit-tests as well.
+
+#include "gtest/gtest.h"
+#include "ngraph/ngraph.hpp"
+#include "util/all_close.hpp"
+#include "util/all_close_f.hpp"
+#include "util/ndarray.hpp"
+#include "util/test_control.hpp"
+#include "util/test_tools.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+static string s_manifest = "${MANIFEST}";
+
+// Combined ops test
+NGRAPH_TEST(${BACKEND_NAME}, mlir_dot_add)
```

Comment:
I would remove the `mlir_` prefix since these tests are really more general than that. Also remove the `_mlir` from the file name.

Reply:
These tests are tailored towards certain features in MLIR end-to-end compilation (e.g. sub-graph extraction, compiling multiple opcodes), and possibly to test MLIR specific optimizations in the future. I did that on purpose to be able to run all mlir tests `--gtest_filter=CPU.mlir_*` . Unless you have a strong opinion on this, I would rather keep it.

Reply:
Are they also running when MLIR is disabled?

Reply:
It sounds like these tests are not named properly. This is not `mlir_dot_add` but is something more like `mlir_single_subgraph` or something like that. The fact that the test uses Dot and Add is irrelevant to the test. You don't need to rename this, I'm just saying the name does not seem to be correct.

Reply:
@dcaballe The tests will run with and without MLIR. 
@rkimballn1 Dot and Add are the currently supported ops :) In the future, these tests will actually go away when we have more ops (so existing unit-tests will exercise MLIR pipeline). For now, I just want to make sure that these tests pass with MLIR enabled before any MLIR check-in.

Reply:
I'm just worried of adding more time to CI when we are not actually testing anything relevant if MLIR is not enabled. Also the confusion this might create: "why my (let's say) PlaidML change is breaking mlir_* tests if MLIR is disable?". However, I don't know about previous precedents/options in this regard so maybe @rkimballn1 or @diyessi can provide better feedback.

Reply:
@nmostafa Add this to the test CMakeLists.txt file around line 175
```
if(NGRAPH_MLIR_ENABLE)
    list(APPEND MULTI_TEST_SRC backend_mlir.in.cpp)
endif()
```
This will only add your new test file if MLIR is enabled, making @dcaballe happy :)

Reply:
Thanks for the hint ! I was wondering myself :) Will do

Reply:
Nice, thanks!